src/core/client.ts

```diff
@@ -877,7 +897,87 @@ export default class SlashAuthClient {
     const authResult = await runMagicLinkLoginIframe(url, this.domainUrl, {
       email: options.email,
     });
+    if (authResult.state !== stateIn) {
+      throw new Error('Invalid state');
+    }
+
+    const tokenResult = await oauthToken({
+      audience: 'default',
+      scope: this.defaultScope,
+      baseUrl: this.domainUrl,
+      client_id: this.options.clientID,
+      code_verifier,
+      code: authResult.code,
+      grant_type: 'authorization_code',
+      redirect_uri: params.redirect_uri,
+      useFormData: false,
+      timeout: this.httpTimeoutMs,
+    });
+    const decodedToken = await this._verifyIdToken(
+      tokenResult.id_token,
+      nonceIn
+    );
+    const cacheEntry = {
+      ...tokenResult,
+      decodedToken,
+      scope: params.scope,
+      audience: 'default',
+      client_id: this.options.clientID,
+    };
+
+    await this.cacheManager.set(cacheEntry);
+
+    this.cookieStorage.save(this.isAuthenticatedCookieName, true, {
+      daysUntilExpire: this.sessionCheckExpiryDays,
+      cookieDomain: this.options.cookieDomain,
+    });
+  }
 
+  public async magicLinkVerify(options: MagicLinkVerifyOptions) {
+    const stateIn = this.continuedInteraction.stateIn;
+    const nonceIn = this.continuedInteraction.nonceIn;
+    const code_verifier = this.continuedInteraction.codeVerifier;
+    const code_challengeBuffer = await sha256(code_verifier);
+    const code_challenge = bufferToBase64UrlEncoded(code_challengeBuffer);
+
+    const hints = {};
+
+    if (options.connectAccounts) {
+      const tokens = (await this.getTokens({
+        detailedResponse: true,
+      })) as GetTokensVerboseResponse;
+
+      if (tokens) {
+        hints['id_token_hint'] = tokens.id_token;
+        hints['login_hint'] = options.email;
+        hints['merge'] = true;
+      }
+    }
+
+    const session = await this.sessionManager.getSession();
+    const params = {
+      client_id: this.options.clientID,
+      redirect_uri: window.location.href,
+      response_type: 'code id_token',
+      code_challenge: code_challenge,
+      code_challenge_method: 'S256',
+      state: stateIn,
+      nonce: nonceIn,
+      hiddenIframe: 'true',
+      response_mode: 'web_message',
+      scope: this.defaultScope,
+      prompt: 'consent',
+      session_id: session.id,
+      continued_interaction_id: this.continuedInteraction.interactionId,
+      ...hints,
+    };
+
+    const url = this._authorizeContinuedUrl(params);
```

Comment:
Do we always want to get a continue URL here? Does this still work with logging in with a magic link by itself?